deltabar/deltabar_lib/deltabar_lib.py

```diff
@@ -129,61 +111,12 @@ def reinitialize_app(self):
       self.data.sectors_available = False  # auto-set to True when available
 
     # Only one sector? We know what sector you are in.
-    if getSectorCount() == 1:
+    if self.track.sector_count == 1:
       self.data.sectors_available = True
 
-    # Click on app area handling - used for toggling modes
-    if not hasattr(self.data, 'click_label'):
-      self.data.click_label = ac.addLabel(self.data.app_id, '')
-      ac.addOnClickedListener(self.data.click_label, sys.modules['deltabar'].onClick)
-    ac.setPosition(self.data.click_label, 0, 0)
-    ac.setSize(self.data.click_label, config.APP_WIDTH, config.APP_HEIGHT)
-
-    # Delta bar main area
-    if not hasattr(self.data, 'bar_area'):
-      self.data.bar_area = ac.addLabel(self.data.app_id, '')
-    ac.setPosition(self.data.bar_area, 0, 0)
-    ac.setSize(self.data.bar_area, config.BAR_WIDTH, config.BAR_HEIGHT)
-    # 0x303030 with 0.65 alpha
-    ac.setBackgroundTexture(self.data.bar_area,
-                            'apps/python/deltabar/background.png')
-
-    # Delta label background area
-    if not hasattr(self.data, 'delta_label_area'):
-      self.data.delta_label_area = ac.addLabel(self.data.app_id, '')
-    ac.setPosition(self.data.delta_label_area,
-                   config.BAR_WIDTH_HALF - config.DELTA_LABEL_WIDTH_HALF,
-                   config.DELTA_LABEL_Y)
-    ac.setSize(self.data.delta_label_area,
-               config.DELTA_LABEL_WIDTH, config.DELTA_LABEL_HEIGHT)
-    # 0x303030 with 0.65 alpha
-    ac.setBackgroundTexture(self.data.delta_label_area,
-                            'apps/python/deltabar/background_delta.png')
-
-    # Delta label text
-    if not hasattr(self.data, 'delta_label'):
-      self.data.delta_label = ac.addLabel(self.data.app_id, '')
-    ac.setPosition(self.data.delta_label,
-                   config.BAR_WIDTH_HALF - config.DELTA_LABEL_WIDTH_HALF,
-                   config.DELTA_LABEL_TEXT_Y)
-    ac.setSize(self.data.delta_label,
-               config.DELTA_LABEL_WIDTH, config.DELTA_LABEL_FONT_SIZE)
-    ac.setFontAlignment(self.data.delta_label, 'center')
-    ac.setFontSize(self.data.delta_label, config.DELTA_LABEL_FONT_SIZE)
-
-    # Banner label (displays current selected mode)
-    if not hasattr(self.data, 'banner_label'):
-      self.data.banner_label = ac.addLabel(self.data.app_id, '')
-    ac.setPosition(self.data.banner_label,
-                   config.BAR_WIDTH_HALF - config.BANNER_TEXT_WIDTH / 2,
-                   config.BANNER_Y)
-    ac.setSize(self.data.banner_label,
-               config.BANNER_TEXT_WIDTH, config.BANNER_FONT_SIZE)
-    ac.setFontAlignment(self.data.banner_label, 'center')
-    ac.setFontSize(self.data.banner_label, config.BANNER_FONT_SIZE)
-    ac.setFontColor(self.data.banner_label, *YELLOW_COLOR)
-
-    track = getTrack()
+    self.ui.reinitialize()
```

Comment:
I'd strongly consider putting this call in the same place as all the other reinitialize() calls happen, down on line 216 in this branch, in acUpdate() "first_update"

if first_update:
  self.ui.reinitialize()
  self.reinitialize_app()
  self.reinitialize_sandbox()